src/lib/functions.ts

```diff
@@ -54,10 +87,17 @@ export async function trigger({
 }) {
   try {
     const res = await functions.get(operationId).run(gleeMessageToFunctionEvent(message, app))
-    let valid = res === undefined
+    const { isValid } = validateData(res, FunctionReturnSchema)
+
+    if ( !isValid ) {
+      logWarningMessage(`Function ${operationId} returned invalid data`, {
+        highlightedWords: [operationId]
+      })
```

Comment:
```suggestion
      const err = new GleeError({
        humanReadableError,
        errors,
      })
      err.message = `Function ${operationId} returned invalid data.`
      logError(err, {
        highlightedWords: [operationId]
      })
      
      
```

You probably have to import GleeError above. If I didn't commit any mistake 😅  that should give a better hint to the user on what actually went wrong.